internal/s/error.go

```diff
@@ -71,6 +77,142 @@ func (err *SupervisorTerminationError) KVs() map[string]interface{} {
 	return acc
 }
 
+// explainLines returns a human-friendly message of the error represented as a slice
+// of lines
+func (err *SupervisorTerminationError) explainLines() []string {
+	// error reporting from children
+	var nodeErrLines []string
+
+	// deal with a single child error
+	if err.nodeErrMap != nil && len(err.nodeErrMap) == 1 {
```

Comment:
You do not really need `err.nodeErrMap != nil`. I am pretty sure `len()` handles that for you.